### PR TITLE
DLP: Added sample for Inspect custom regex

### PR DESCRIPTION
**Pull request by jainamtrivedi-crest**
- Added sample for [Inspect Custom Regex](https://cloud.google.com/dlp/docs/creating-custom-infotypes-regex#dlp-inspect-custom-regex-java) and unit tests

### CODE REVIEW

File: dlp/api/Snippets/InspectDataWithCustomRegex.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+using Google.Cloud.Dlp.V2;
+
+public class InspectDataWithCustomRegex
+{
+    public static InspectContentResponse InspectDataCustomRegex(string projectId, string text, string customRegex, InfoType infoType = null)
+    {
+        // Instantiate a client.
+        var dlp = DlpServiceClient.Create();
+
+        // Construct the byte content item.
```
Comment (ivanmed) [inline dlp/api/Snippets/InspectDataWithCustomRegex.cs:28]: I think there is no need for byteContentItem here, you can just assign the 'text' parameter directly to ContentItem.Value, something like:
var contentItem = new ContentItem { Value = text };

Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/InspectDataWithCustomRegex.cs]: Removed the byte content item. In JAVA code sample, they have used byte content item; So, I used it here as well

**Review by ivanmed — APPROVED** [review]

File: dlp/api/Snippets.Tests/InspectDataWithCustomRegexTests.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        private DlpTestFixture _fixture;
+        public InspectDataWithCustomRegexTests(DlpTestFixture fixture) => _fixture = fixture;
+
+        [Fact]
+        public void TestInspectDataWithCustomRegex()
+        {
+            string mrn = "444-5-22222";
+            string textToInspect = $"Patients MRN {mrn}";
+            string customRegexPattern = "[1-9]{3}-[1-9]{1}-[1-9]{5}";
+            InfoType infoType = new() { Name = "C_MRN" };
```
Comment (amanda-tarafa) [inline dlp/api/Snippets.Tests/InspectDataWithCustomRegexTests.cs:31]: tiny nit: Be consistent, you have been using: 
```suggestion
            var infoType = new InfoType { Name = "C_MRN" };
```

File: dlp/api/Snippets.Tests/InspectDataWithCustomRegexTests.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            var result = InspectDataWithCustomRegex.InspectDataCustomRegex(_fixture.ProjectId, textToInspect, customRegexPattern, infoType);
+            var findings = result.Result.Findings;
+            Assert.Single(findings);
+            Assert.Equal(mrn, findings[0].Quote);
```
Comment (amanda-tarafa) [inline dlp/api/Snippets.Tests/InspectDataWithCustomRegexTests.cs:35]:
```suggestion
            var finding = Assert.Single(findings);
            Assert.Equal(mrn, finding.Quote);
```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
A couple of changed but otherwise looks good.

File: dlp/api/Snippets.Tests/InspectDataWithCustomRegexTests.cs (hunk 2 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/InspectDataWithCustomRegexTests.cs]: Changed.

**Review by amanda-tarafa — APPROVED** [review]
Thanks!

**Pull request merged**